pkg/log/zap/flags.go

```diff
@@ -158,3 +166,34 @@ func (v sampleValue) IsBoolFlag() bool {
 func (v sampleValue) Type() string {
 	return "sample"
 }
+
+type timeformatValue struct {
+	set bool
+	str string
+}
+
+func (v *timeformatValue) Set(s string) error {
+	v.set = true
+	if len(s) > 1 {
+		if s == "unix" || s == "iso8601" {
```

Comment:
While we're at it, can we handle everything else zap supports? See https://github.com/uber-go/zap/blob/8a2ee5670ced5d94154bf385dc6a362722945daf/zapcore/encoder.go#L121-L133

Reply:
I would have liked to have useds the timeEncoder.UnmarshalText here, but it doesn't seem possible (or I can't figure out how). So for now, it can support any timeformat that zap currently supports. If they add or remove timeformats in future values, there might be problems.

Reply:
The `UnmarshalText` function is a great idea. It looks like this should work:

```go
func (v *timeformatValue) Set(s string) error {
	v.set = true
	v.timeEncoder.UnmarshalText([]byte(s))
	return nil
}
```